site/content/troubleshooting-guide/index.md

```diff
@@ -67,4 +67,34 @@ AWS.Deploy.Tools, internally uses a variety of different services to host your .
 
 **Why is this happening**: This happens due to a know issue with the AWS Cloud Development Kit (CDK). The CDK is used to AWS.Deploy.Tools under the covers and it cannot cannot access the `$TEMP` directory inside the `$USERPROFILE` path if it contains a whitespace character.
 
-**Resolution**: See [here](../../docs/getting-started/custom-workspace) for guidance on setting a custom workspace that will be used by AWS.Deploy.tools.
\ No newline at end of file
+**Resolution**: See [here](../../docs/getting-started/custom-workspace) for guidance on setting a custom workspace that will be used by AWS.Deploy.tools.
+
+## AWS CDK Bootstrap related Deployment Failure
+
+A common error that is displayed for CDK Bootstrap related deployment failures is the following:
+```
+The AWS CDK Bootstrap, which is the process of provisioning initial resources for the deployment environment, has failed. Please review the output above for additional details [and check out our troubleshooting guide for the most common failure reasons]. You can learn more about CDK bootstrapping at https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html.
+```
+The .NET Deployment Tool uses [AWS CDK](https://docs.aws.amazon.com/cdk/v2/guide/home.html) to create the AWS infrastructure needed to deploy your application. The AWS CDK is a framework for defining cloud infrastructure in code and provisioning it through AWS CloudFormation. 
+
+Deploying AWS CDK apps into an AWS environment (a combination of an AWS account and region) requires that you provision resources the AWS CDK needs to perform the deployment. These resources include an Amazon S3 bucket for storing files and IAM roles that grant permissions needed to perform deployments. The process of provisioning these initial resources is called [bootstrapping](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html).
+
+The required resources are defined in a AWS CloudFormation stack, called the bootstrap stack, which is usually named `CDKToolkit`. Like any AWS CloudFormation stack, it appears in the AWS CloudFormation console once it has been deployed.
+
+There could be several reasons why you are experiencing this issue. However, the most common ones are related to **Insufficient IAM Permissions** and an **Existing CDK Staging Bucket**. 
+
+**Possible Failure Reason**: When you try to deploy your application using the .NET Deployment Tool, your AWS account is checked for an existing `CDKToolkit` CloudFormation bucket. If it exists with the appropriate version, we continue with the deployment. If none exists, or you have an older version of it, then we try to create it or update the existing one. During that process, a set of IAM permissions are needed which typically include CloudFormation, S3 and ECR permissions. If you are missing some IAM permissions, that will cause the CDK Bootstrap to fail.
```

Comment:
Replace with 
**Insufficient IAM Permissions** - CDKBoostrap failed because your profile does not have sufficient permissions to create the boostrap stack.  Check the log - in this case, you should see an error that looks something like this:

```
LookupRole API: iam:GetRole User: arn:aws:iam::123456789101:user/user is not authorized to perform: iam:GetRole on resource
```

Reply:
Done